addons/account/models/partner.py

```diff
@@ -496,6 +503,18 @@ def _compute_bank_count(self):
         for partner in self:
             partner.bank_account_count = mapped_data.get(partner.id, 0)
 
+    @api.depends('bank_ids')
+    def _compute_duplicated_bank_account_partners_count(self):
+        for partner in self:
+            if not partner.bank_ids:
+                partner.duplicated_bank_account_partners_count = 0
+            else:
+                domain = []
+                for bank in partner.bank_ids:
+                    domain = expression.OR([domain, [('acc_number', '=', bank.acc_number), ('bank_id', '=', bank.bank_id.id)]])
```

Comment:
This is leading to a very complicated domain for nothing.
```py
domains = []
for bank in partner.bank_ids:
    domains.append([('acc_number', '=', bank.acc_number), ('bank_id', '=', bank.bank_id.id)])
domain = expression.OR(domains)
domain = expression.AND([domain, [('partner_id', '!=', partner.id or partner._origin.id)]])
```
...is better because you are making a OR only once

Reply:
`partner.id or partner._origin.id` is equivalent to `partner._origin.id`